File: solutions/csharp/darts/2/Darts.cs
using System;

public static class Darts
{
    public static int Score(double x, double y) =>
        DistanceFromCenter(x, y) switch
        {
            > 10.0 => 0,
            > 5.0 => 1,
            > 1.0 => 5,
            _ => 10,
        };

    public static double DistanceFromCenter(double x, double y) =>
        Math.Sqrt(x * x + y * y);
}

File: solutions/typescript/darts/2/darts.ts
export function score(x: number, y: number): number {
  const dist = Math.sqrt(x**2 + y**2);
  if (dist <= 1)
    return 10;
  else if (dist <= 5)
    return 5;
  else if (dist <= 10)
    return 1;
  else
    return 0;
}
